examples/pipeline_wavernn/wavernn.py

```diff
@@ -0,0 +1,586 @@
+        default=-100,
+        type=float,
+        help="the min db value for spectrogam normalization",
+    )
+    parser.add_argument(
+        "--n-res-block",
+        default=10,
+        type=int,
+        help="the number of ResBlock in stack",
+    )
+    parser.add_argument(
+        "--n-rnn",
+        default=512,
+        type=int,
+        help="the dimension of RNN layer",
+    )
+    parser.add_argument(
+        "--n-fc",
+        default=512,
+        type=int,
+        help="the dimension of fully connected layer",
+    )
+    parser.add_argument(
+        "--kernel-size",
+        default=5,
+        type=int,
+        help="the number of kernel size in the first Conv1d layer",
+    )
+    parser.add_argument(
+        "--n-freq",
+        default=80,
+        type=int,
+        help="the number of bins in a spectrogram",
+    )
+    parser.add_argument(
+        "--n-hidden",
+        default=128,
+        type=int,
+        help="the number of hidden dimensions",
+    )
+    parser.add_argument(
+        "--n-output",
+        default=128,
+        type=int,
+        help="the number of output dimensions",
+    )
+    parser.add_argument(
+        "--mode",
+        default="waveform",
+        choices=["waveform", "mol"],
+        type=str,
+        help="the type of waveform",
+    )
+    parser.add_argument(
+        "--seq-len-factor",
+        default=5,
+        type=int,
+        help="seq_length = hop_length * seq_len_factor",
+    )
+    parser.add_argument(
+        "--test-samples",
+        default=50,
+        type=float,
+        help="the number of waveforms for testing",
+    )
+    parser.add_argument(
+        "--file-path",
+        default="/private/home/jimchen90/datasets/LJSpeech-1.1/wavs/",
+        type=str,
+        help="the path of audio files",
+    )
+
+    args = parser.parse_args()
+    return args
+
+
+def signal_handler(a, b):
+    global SIGNAL_RECEIVED
+    print("Signal received", a, datetime.now().strftime("%y%m%d.%H%M%S"), flush=True)
+    SIGNAL_RECEIVED = True
+
+
+def train_one_epoch(
+    model, mode, criterion, optimizer, scheduler, data_loader, device, epoch
+):
+
+    model.train()
+
+    sums = defaultdict(lambda: 0.0)
+    start1 = time()
+
+    metric = MetricLogger("train_iteration")
+    metric("epoch", epoch)
+
+    for waveform, specgram, target in bg_iterator(data_loader, maxsize=2):
+
+        start2 = time()
+
+        waveform = waveform.to(device)
+        specgram = specgram.to(device)
+        target = target.to(device)
+
+        output = model(waveform, specgram)
+        output, target = output.squeeze(1), target.squeeze(1)
+
+        if mode == 'waveform':
+            output = output.transpose(1, 2)
+            target = target.long()
+
+        elif mode == 'mol':
+            target = target.unsqueeze(-1)
+
+        else:
+            raise ValueError(
+                f"Expected mode: `waveform` or `mol`, but found {mode}"
+            )
```

Comment:
I'd suggest using mol as the default value.
```python
        if mode == 'waveform':
            output = output.transpose(1, 2)
            target = target.long()

        else:
            # use mol mode
            target = target.unsqueeze(-1)
```

Reply:
Good point. I have updated it.